File: Estruturais/Proxy/proxy.py
# objeto real ------------------------------------
class ServicoReal:
    def executar(self):
        print('serviço real')

class ProxyServicoReal:
    def __init__(self):
        self.servicoReal = ServicoReal() 

    def executar(self):
        print('Antes')
        self.servicoReal.executar()
        print('Depois')   

proxy = ProxyServicoReal()
proxy.executar()                

File: Estruturais/Proxy/proxy.js
// objeto real ---------------------------------------------------
class ServicoReal{
    executar(){
        console.log('serviço real')
    }
}

//proxy ---------------------------------------------------
class ProxyServicoReal{
    constructor(){
        this.servicoReal = new ServicoReal();

    }
    executar(){
        console.log('Antes de executar a Proxy');
        this.servicoReal.executar();
        console.log('depois de executar a proxy');
    }
}

//uso do proxy ---------------------------------------------------
const proxy = new ProxyServicoReal();
proxy.executar();